tools/src/main/scala/scala/scalanative/nir/parser/NirParser.scala

```diff
@@ -0,0 +1,451 @@
+package scala.scalanative
+package nir
+package parser
+
+import fastparse.WhitespaceApi
+import fastparse.noApi._
+
+trait NirParser[T] {
+  def parser: P[T]
+
+  final def apply(nir: String) =
+    parser.parse(nir)
+}
+
+object Base {
+  import fastparse.all._
+  def byte: P[Byte]   = P("-".!.? ~ CharIn('0' to '9').rep(1).! map { case (None, n) => n.toByte; case (_, n) => ("-" + n).toByte })
```

Comment:
If I'm not mistaken, for fastparse it's idiomatic to have parsers defined as vals whenever possible, not defs. This ensures we don't instantiate them over and over again for no good reason.